pyfstat/core.py

```diff
@@ -1649,57 +1649,46 @@ def get_semicoherent_twoF(
             self.whatToCompute,
         )
 
-        # if not self.transientWindowType:
-        #    if self.BSGL is False:
-        #        return self.FstatResults.twoF[0]
-        #    twoF = np.float(self.FstatResults.twoF[0])
-        #    self.twoFX[0] = self.FstatResults.twoFPerDet(0)
-        #    self.twoFX[1] = self.FstatResults.twoFPerDet(1)
-        #    log10_BSGL = lalpulsar.ComputeBSGL(twoF, self.twoFX,
-        #                                       self.BSGLSetup)
-        #    return log10_BSGL/np.log10(np.exp(1))
+        twoF_per_segment = self._get_per_segment_twoF()
+        twoF = twoF_per_segment.sum()
 
-        det_stat_per_segment = self._get_per_segment_det_stat()
-        detStat = det_stat_per_segment.sum()
-
-        if np.isnan(detStat):
-            logging.debug("NaNs in semi-coherent detection statistic treated as zero")
-            det_stat_per_segment = np.nan_to_num(det_stat_per_segment, nan=0.0)
-            detStat = det_stat_per_segment.sum()
+        if np.isnan(twoF):
+            logging.debug(
+                "NaNs in per-segment 2F treated as zero"
+                " and semi-coherent 2F re-computed."
+            )
+            twoF_per_segment = np.nan_to_num(twoF_per_segment, nan=0.0)
+            twoF = twoF_per_segment.sum()
 
         if record_segments:
-            self.detStat_per_segment = det_stat_per_segment
-
-        return detStat
-
-    def _get_per_segment_det_stat(self):
-        FS = lalpulsar.ComputeTransientFstatMap(
-            self.FstatResults.multiFatoms[0], self.semicoherentWindowRange, False
-        )
+            self.detStat_per_segment = twoF_per_segment
```

Comment:
Should we re-name this variable to self.twoF_per_segment since there's no BSGL per segment?

Reply:
I was thinking the same. It doesn't actually seem to be used anywhere in the package or examples, either, so I guess it won't break anything and is purely an undocumented debug feature anyway.

Reply:
...it was used in a test after all. Fixed now.